### PR TITLE
Feat/az/update

### LINKED ISSUE #9: Oppdatere favicon og breadcrums
Favicon med ny digdir-logo mangler, og teksten i breadcrums er mye mindre i brukervilkår enn resten av løsningen

- [ ] Oppdatere favicon med ny digdir-logo
- [ ] Oppdatere breadcrums så den ser lik ut som resten av registrering

### PR DESCRIPTION
**Pull request by azaxarov**
Resolves #9

### CODE REVIEW

File: src/providers/auth/index.tsx [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
     try {
-      await service.init(true);
+      const authenticated = await service.init();
+
       setIsInitialised(true);
-    } catch (error) {
-      console.error(error);
+      setIsAuthenticated(authenticated);
+    } catch (e) {
+      // TODO: handle service errors and log them to Sentry
```
Comment (matsjs) [inline src/providers/auth/index.tsx:19]: samme her

File: src/services/auth/index.ts [hunk 2]
```diff
@@ -1,151 +1,199 @@
...
-    if (!!loginRequired && !this.isAuthenticated()) {
-      await this.login();
+      await this.cleanUp();
+
+      this.user = await this.manager.signinSilent();
+    } catch (e) {
+      if (e.error === OidcError.LOGIN_REQUIRED) {
+        await this.signIn();
+      }
+      // TODO: handle service errors and log them to Sentry
```
Comment (matsjs) [inline src/services/auth/index.ts:36]: samme her

File: src/services/auth/index.ts [hunk 3]
```diff
@@ -1,151 +1,199 @@
...
 
-  public async login(): Promise<void> {
+  public async signIn(): Promise<void> {
     try {
-      await this.kc.login();
-    } catch (error) {
-      console.error(error);
+      await this.manager.signinRedirect({ data: { path: location.href } });
+    } catch (e) {
+      // TODO: handle service errors and log them to Sentry
```
Comment (matsjs) [inline src/services/auth/index.ts:46]: samme her

File: src/services/auth/index.ts [hunk 4]
```diff
@@ -1,151 +1,199 @@
...
+  public async signOut(): Promise<void> {
     try {
-      await this.kc.logout({
-        redirectUri: this.configuration.logoutRedirectUri
-      });
-    } catch (error) {
-      console.error(error);
+      await this.manager.signoutRedirect();
+    } catch (e) {
+      // TODO: handle service errors and log them to Sentry
```
Comment (matsjs) [inline src/services/auth/index.ts:54]: samme her

File: src/services/auth/index.ts [hunk 5]
```diff
@@ -1,151 +1,199 @@
...
+      if (this.isTokenExpired()) {
+        this.user = await this.manager.signinSilent();
+      }
+
+      if (this.user && this.isAuthenticated() && !this.isTokenExpired()) {
+        const { token_type, access_token } = this.user;
+        return `${token_type} ${access_token}`;
+      }
+    } catch (e) {
+      // TODO: handle service errors and log them to Sentry
```
Comment (matsjs) [inline src/services/auth/index.ts:77]: samme her

**Review by matsjs — APPROVED** [review]

File: src/entrypoints/auth/index.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  try {
+    const user = await (isInIframe
+      ? manager.signinSilentCallback()
+      : manager.signinRedirectCallback());
+
+    if (!isInIframe && user?.state?.path) {
+      path = user.state.path.replace(location.origin, '');
+    }
+  } catch (e) {
+    // TODO: handle errors and log them to Sentry
```
[the remark this answers is not in the document]
Comment (azaxarov) [reply src/entrypoints/auth/index.ts:29]: Fikser i en annen issue

**Pull request merged**